components/vision.py

```diff
@@ -29,8 +37,36 @@ class Vision:
     fuse_vision_observations = tunable(True)
     gate_innovation = tunable(True)
 
+    def _camera_to_robot(self) -> Transform2d:
+        turret_azimuth = self.turret.get_angle()
+        return Transform2d(
+            -(self.TURRET_OFFSET + math.cos(turret_azimuth) * self.CAMERA_OFFSET),
+            -(math.sin(turret_azimuth) * self.CAMERA_OFFSET),
+            -turret_azimuth,
+        )
+
     def __init__(self) -> None:
         self.camera = PhotonCamera("gloworm")
+
+        if wpilib.RobotBase.isSimulation():
+            self.sim_vision_system = SimVisionSystem(
+                "gloworm",
+                75.76079874010732,
+                math.degrees(self.CAMERA_PITCH),
+                Transform2d(0, 0, 0),
+                self.CAMERA_HEIGHT,
+                10.0,
+                1280,
+                720,
+                0.0,
+            )
+            self.sim_vision_system.addSimVisionTarget(
+                SimVisionTarget(
+                    Pose2d(0, 0, 0), self.TARGET_HEIGHT, self.GOAL_RADIUS * 2, 0.05
```

Comment:
It'd be worth using keyword arguments for the constants here, so it's clearer what they are.